core/main/src/firebolt/handlers/internal_rpc.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Comcast Cable Communications Management, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use jsonrpsee::{core::RpcResult, proc_macros::rpc, RpcModule};
+use ripple_sdk::{
+    api::{
+        apps::{AppEvent, AppManagerResponse, AppMethod, AppRequest, AppResponse},
+        firebolt::{fb_general::ListenRequestWithEvent, fb_telemetry::TelemetryPayload},
+        gateway::rpc_gateway_api::CallContext,
+    },
+    async_trait::async_trait,
+    log::error,
+    tokio::sync::oneshot,
+};
+
+use crate::{
+    firebolt::rpc::RippleRPCProvider,
+    service::{apps::app_events::AppEvents, telemetry_builder::TelemetryBuilder},
+    state::platform_state::PlatformState,
+    utils::rpc_utils::rpc_await_oneshot,
+};
+
+#[rpc(server)]
+pub trait Internal {
```

Comment:
If we're defining a trait why don't we need it defined in the SDK instead of here?

Reply:
It will be used internally within extensions not exposed as an open rpc to apps